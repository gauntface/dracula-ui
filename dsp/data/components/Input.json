{
  "class": "component",
  "type": "page",
  "id": "Input",
  "name": "Input",
  "last_updated": "2021-03-16T23:22:40.224Z",
  "last_updated_by": "System",
  "description": "Input is a styled HTML Input.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep inputs accessible.",
  "related_entity_ids": [],
  "tags": ["component"],
  "snippets": {
    "trigger": "input",
    "languages": {
      "javascriptreact": "<Input placeholder=\"Placeholder\" color=\"white\" />;\n",
      "html": "<input\n  placeholder=\"Placeholder\"\n  color=\"white\"\n  class=\"drac-input drac-input-white drac-text-white\"\n/>\n"
    }
  },
  "ext_com_draculaui_variations": [
    {
      "title": "Usage",
      "html": "<input\n  placeholder=\"Placeholder\"\n  color=\"white\"\n  class=\"drac-input drac-input-white drac-text-white\"\n/>\n",
      "react": "<Input placeholder=\"Placeholder\" color=\"white\" />;\n",
      "docs": "",
      "screenshot": "./dsp/assets/Input.png",
      "svgData": "./dsp/assets/svgs/Input.svg"
    },
    {
      "title": "Colors",
      "html": "<div style=\"display: flex; flex-direction: column\">\n  <input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"white\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-white drac-text-white\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"cyan\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-cyan drac-text-cyan\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"green\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-green drac-text-green\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"orange\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-orange drac-text-orange\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"pink\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-pink drac-text-pink\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"purple\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-purple drac-text-purple\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"red\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-red drac-text-red\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"yellow\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-yellow drac-text-yellow\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"white\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-white drac-text-white\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"cyan\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-cyan drac-text-cyan\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"green\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-green drac-text-green\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"orange\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-orange drac-text-orange\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"pink\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-pink drac-text-pink\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"purple\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-purple drac-text-purple\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"red\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-red drac-text-red\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"yellow\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-yellow drac-text-yellow\"\n  />\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"column\" }}>\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"white\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"cyan\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"green\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"orange\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"pink\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"purple\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"red\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"yellow\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"white\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"cyan\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"green\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"orange\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"pink\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"purple\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"red\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"yellow\"\n    style={{ marginBottom: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/InputColors.png",
      "svgData": "./dsp/assets/svgs/InputColors.svg"
    },
    {
      "title": "Sizes",
      "html": "<div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      placeholder=\"Placeholder\"\n      color=\"white\"\n      class=\"drac-input drac-input-lg drac-input-white drac-text-white\"\n    />\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      placeholder=\"Placeholder\"\n      color=\"white\"\n      class=\"drac-input drac-input-white drac-text-white\"\n    />\n  </div>\n  <div style=\"margin-bottom: 10px\">\n    <input\n      placeholder=\"Placeholder\"\n      color=\"white\"\n      class=\"drac-input drac-input-sm drac-input-white drac-text-white\"\n    />\n  </div>\n</div>\n",
      "react": "<div>\n  <div style={{ marginBottom: 10 }}>\n    <Input placeholder=\"Placeholder\" size=\"large\" color=\"white\" />\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Input placeholder=\"Placeholder\" size=\"medium\" color=\"white\" />\n  </div>\n  <div style={{ marginBottom: 10 }}>\n    <Input placeholder=\"Placeholder\" size=\"small\" color=\"white\" />\n  </div>\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/InputSizes.png",
      "svgData": "./dsp/assets/svgs/InputSizes.svg"
    },
    {
      "title": "Variants",
      "html": "<div style=\"display: flex; flex-direction: column\">\n  <input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"white\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-white drac-text-white\"\n  /><input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"white\"\n    style=\"margin-bottom: 20px\"\n    class=\"drac-input drac-input-outline drac-input-white drac-text-white\"\n  />\n</div>\n",
      "react": "<div style={{ display: \"flex\", flexDirection: \"column\" }}>\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"normal\"\n    color=\"white\"\n    style={{ marginBottom: 20 }}\n  />\n  <Input\n    placeholder=\"Placeholder\"\n    variant=\"outline\"\n    color=\"white\"\n    style={{ marginBottom: 20 }}\n  />\n</div>;\n",
      "docs": "",
      "screenshot": "./dsp/assets/InputVariants.png",
      "svgData": "./dsp/assets/svgs/InputVariants.svg"
    }
  ],
  "ext_com_draculaui_props": {
    "color": {
      "defaultValue": null,
      "description": "The Dracula UI theme color to be used",
      "name": "color",
      "parent": {
        "fileName": "src/components/Input/Input.tsx",
        "name": "InputProps"
      },
      "declarations": [
        { "fileName": "src/components/Input/Input.tsx", "name": "InputProps" }
      ],
      "required": false,
      "type": {
        "name": "\"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | undefined"
      }
    },
    "size": {
      "defaultValue": null,
      "description": "Controls the size of the input based on pre-configured Dracula UI sizes.\nOptions: `small`, `medium`, `large`",
      "name": "size",
      "parent": {
        "fileName": "src/components/Input/Input.tsx",
        "name": "InputProps"
      },
      "declarations": [
        { "fileName": "src/components/Input/Input.tsx", "name": "InputProps" }
      ],
      "required": false,
      "type": { "name": "\"large\" | \"medium\" | \"small\" | undefined" }
    },
    "variant": {
      "defaultValue": null,
      "description": "Controls the variation the input.\n`normal` -> Regular Input component with a light background color.\n`outline` -> Keeps the accent color, but removes the background.",
      "name": "variant",
      "parent": {
        "fileName": "src/components/Input/Input.tsx",
        "name": "InputProps"
      },
      "declarations": [
        { "fileName": "src/components/Input/Input.tsx", "name": "InputProps" }
      ],
      "required": false,
      "type": { "name": "\"normal\" | \"outline\" | undefined" }
    },
    "p": {
      "defaultValue": null,
      "description": "",
      "name": "p",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "py": {
      "defaultValue": null,
      "description": "",
      "name": "py",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "px": {
      "defaultValue": null,
      "description": "",
      "name": "px",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pt": {
      "defaultValue": null,
      "description": "",
      "name": "pt",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pb": {
      "defaultValue": null,
      "description": "",
      "name": "pb",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pl": {
      "defaultValue": null,
      "description": "",
      "name": "pl",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "pr": {
      "defaultValue": null,
      "description": "",
      "name": "pr",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "m": {
      "defaultValue": null,
      "description": "",
      "name": "m",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "my": {
      "defaultValue": null,
      "description": "",
      "name": "my",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mx": {
      "defaultValue": null,
      "description": "",
      "name": "mx",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mt": {
      "defaultValue": null,
      "description": "",
      "name": "mt",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mb": {
      "defaultValue": null,
      "description": "",
      "name": "mb",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "ml": {
      "defaultValue": null,
      "description": "",
      "name": "ml",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    },
    "mr": {
      "defaultValue": null,
      "description": "",
      "name": "mr",
      "declarations": [
        { "fileName": "dracula-ui/src/base/spacing.ts", "name": "TypeLiteral" }
      ],
      "required": false,
      "type": {
        "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
      }
    }
  },
  "ext_com_draculaui_docgen": {
    "tags": {},
    "description": "Input is a styled HTML Input.\n\nThere are no behavior changes applied to the native HTML tag other\nthan light styling done via CSS in order to keep inputs accessible.",
    "displayName": "Input",
    "methods": [],
    "props": {
      "color": {
        "defaultValue": null,
        "description": "The Dracula UI theme color to be used",
        "name": "color",
        "parent": {
          "fileName": "src/components/Input/Input.tsx",
          "name": "InputProps"
        },
        "declarations": [
          { "fileName": "src/components/Input/Input.tsx", "name": "InputProps" }
        ],
        "required": false,
        "type": {
          "name": "\"purpleCyan\" | \"yellowPink\" | \"cyanGreen\" | \"pinkPurple\" | \"animated\" | \"cyan\" | \"green\" | \"orange\" | \"pink\" | \"purple\" | \"red\" | \"yellow\" | \"white\" | \"black\" | \"blackSecondary\" | \"blackLight\" | undefined"
        }
      },
      "size": {
        "defaultValue": null,
        "description": "Controls the size of the input based on pre-configured Dracula UI sizes.\nOptions: `small`, `medium`, `large`",
        "name": "size",
        "parent": {
          "fileName": "src/components/Input/Input.tsx",
          "name": "InputProps"
        },
        "declarations": [
          { "fileName": "src/components/Input/Input.tsx", "name": "InputProps" }
        ],
        "required": false,
        "type": { "name": "\"large\" | \"medium\" | \"small\" | undefined" }
      },
      "variant": {
        "defaultValue": null,
        "description": "Controls the variation the input.\n`normal` -> Regular Input component with a light background color.\n`outline` -> Keeps the accent color, but removes the background.",
        "name": "variant",
        "parent": {
          "fileName": "src/components/Input/Input.tsx",
          "name": "InputProps"
        },
        "declarations": [
          { "fileName": "src/components/Input/Input.tsx", "name": "InputProps" }
        ],
        "required": false,
        "type": { "name": "\"normal\" | \"outline\" | undefined" }
      },
      "p": {
        "defaultValue": null,
        "description": "",
        "name": "p",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "py": {
        "defaultValue": null,
        "description": "",
        "name": "py",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "px": {
        "defaultValue": null,
        "description": "",
        "name": "px",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pt": {
        "defaultValue": null,
        "description": "",
        "name": "pt",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pb": {
        "defaultValue": null,
        "description": "",
        "name": "pb",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pl": {
        "defaultValue": null,
        "description": "",
        "name": "pl",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "pr": {
        "defaultValue": null,
        "description": "",
        "name": "pr",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "m": {
        "defaultValue": null,
        "description": "",
        "name": "m",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "my": {
        "defaultValue": null,
        "description": "",
        "name": "my",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mx": {
        "defaultValue": null,
        "description": "",
        "name": "mx",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mt": {
        "defaultValue": null,
        "description": "",
        "name": "mt",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mb": {
        "defaultValue": null,
        "description": "",
        "name": "mb",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "ml": {
        "defaultValue": null,
        "description": "",
        "name": "ml",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      },
      "mr": {
        "defaultValue": null,
        "description": "",
        "name": "mr",
        "declarations": [
          {
            "fileName": "dracula-ui/src/base/spacing.ts",
            "name": "TypeLiteral"
          }
        ],
        "required": false,
        "type": {
          "name": "\"none\" | \"xxs\" | \"xs\" | \"sm\" | \"md\" | \"lg\" | undefined"
        }
      }
    }
  }
}
