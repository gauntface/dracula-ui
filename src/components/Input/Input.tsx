import { ColorMap } from '@/base/colors'
import React, { HTMLAttributes } from 'react'

export const inputVariants = {
  normal: 'drac-input',
  outline: 'drac-input-outline'
}

export const inputColors: ColorMap = {
  black: 'drac-input-black drac-text-white',
  white: 'drac-input-white drac-text-white',
  cyan: 'drac-input-cyan drac-text-cyan',
  green: 'drac-input-green drac-text-green',
  orange: 'drac-input-orange drac-text-orange',
  pink: 'drac-input-pink drac-text-pink',
  purple: 'drac-input-purple drac-text-purple',
  red: 'drac-input-red drac-text-red',
  yellow: 'drac-input-yellow drac-text-yellow',
  animated: '',
  cyanGreen: '',
  pinkPurple: '',
  purpleCyan: '',
  yellowPink: ''
}

export interface InputProps extends HTMLAttributes<HTMLInputElement> {
  variant: keyof typeof inputVariants
  color: keyof typeof inputColors
}

export const Input: React.FC<InputProps> = (props) => {
  const finalProps = {
    ...props,
    className: `drac-input drac-text drac-text-md ${
      inputVariants[props.variant]
    } ${inputColors[props.color]}`
  }

  return <input {...finalProps} />
}

Input.displayName = 'Input'
