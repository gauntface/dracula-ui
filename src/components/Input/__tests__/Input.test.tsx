import React from 'react'
import * as ReactDOM from 'react-dom'
import { Input } from '../Input'

describe('Input', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Input variant="outline" color="white" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})
