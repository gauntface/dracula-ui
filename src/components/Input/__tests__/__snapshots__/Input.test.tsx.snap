// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: Input Basic Usage 1`] = `
Object {
  "docs": "Styles or abstracts HTML input elements.",
  "html": "<input
  placeholder=\\"Input\\"
  color=\\"white\\"
  class=\\"drac-input drac-input-white drac-text-white\\"
/>
",
  "react": "<Input placeholder=\\"Input\\" color=\\"white\\" />;
",
  "title": "Usage",
}
`;

exports[`Site: Input Colors 1`] = `
Object {
  "docs": "Inputs can be customized to use any of the Dracula UI theme colors.",
  "html": "<div class=\\"drac-box\\">
  <input
    color=\\"purple\\"
    placeholder=\\"purple\\"
    m=\\"xs\\"
    class=\\"drac-input drac-input-purple drac-text-purple drac-m-xs\\"
  /><input
    color=\\"green\\"
    placeholder=\\"green\\"
    m=\\"xs\\"
    class=\\"drac-input drac-input-green drac-text-green drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input color=\\"purple\\" placeholder=\\"purple\\" m=\\"xs\\" />
  <Input color=\\"green\\" placeholder=\\"green\\" m=\\"xs\\" />
</Box>;
",
  "title": "Colors",
}
`;

exports[`Site: Input Sizes 1`] = `
Object {
  "docs": "Inputs can be customize to use several different sizes.",
  "html": "<div class=\\"drac-box\\">
  <input
    placeholder=\\"small\\"
    m=\\"xs\\"
    class=\\"drac-input drac-input-sm drac-m-xs\\"
  /><input placeholder=\\"medium\\" m=\\"xs\\" class=\\"drac-input drac-m-xs\\" /><input
    placeholder=\\"large\\"
    m=\\"xs\\"
    class=\\"drac-input drac-input-lg drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input size=\\"small\\" placeholder=\\"small\\" m=\\"xs\\" />
  <Input size=\\"medium\\" placeholder=\\"medium\\" m=\\"xs\\" />
  <Input size=\\"large\\" placeholder=\\"large\\" m=\\"xs\\" />
</Box>;
",
  "title": "Sizes",
}
`;

exports[`Site: Input Variants 1`] = `
Object {
  "docs": "Use the \`outline\` variant to represent subtler text Inputs",
  "html": "<div class=\\"drac-box\\">
  <input
    variant=\\"normal\\"
    placeholder=\\"normal\\"
    m=\\"xs\\"
    class=\\"drac-input drac-m-xs\\"
  /><input
    variant=\\"outline\\"
    placeholder=\\"outline\\"
    m=\\"xs\\"
    class=\\"drac-input drac-input-outline drac-m-xs\\"
  />
</div>
",
  "react": "<Box>
  <Input variant=\\"normal\\" placeholder=\\"normal\\" m=\\"xs\\" />
  <Input variant=\\"outline\\" placeholder=\\"outline\\" m=\\"xs\\" />
</Box>;
",
  "title": "Variants",
}
`;
